File: sampleApp/src/main/java/com/nytimes/android/external/playbillingtester/sample/EmptyItemDecoration.java
package com.nytimes.android.external.playbillingtester.sample;

import android.graphics.Rect;
import android.support.v7.widget.RecyclerView;
import android.view.View;

public class EmptyItemDecoration extends RecyclerView.ItemDecoration {

    private final int verticalSize;

    public EmptyItemDecoration(int size) {
        this.verticalSize = size;
    }

    @Override
    public void getItemOffsets(Rect outRect, View view, RecyclerView parent,
                               RecyclerView.State state) {
        if (parent.getChildAdapterPosition(view) != parent.getAdapter().getItemCount() - 1) {
            outRect.bottom = verticalSize;
        }
    }
}

File: sampleApp/src/main/java/com/nytimes/android/external/register/sample/EmptyItemDecoration.kt
package com.nytimes.android.external.register.sample

import android.graphics.Rect
import androidx.recyclerview.widget.RecyclerView
import android.view.View

class EmptyItemDecoration(private val verticalSize: Int) : RecyclerView.ItemDecoration() {

    override fun getItemOffsets(outRect: Rect, view: View, parent: RecyclerView, state: RecyclerView.State) {
        if (parent.getChildAdapterPosition(view) != parent.adapter?.itemCount ?: 0 - 1) {
            outRect.bottom = verticalSize
        }
    }
}
